memtable/write_buffer_manager.cc

```diff
@@ -115,52 +147,26 @@ void WriteBufferManager::FreeMemWithCache(size_t mem) {
   s.PermitUncheckedError();
 }
 
-void WriteBufferManager::BeginWriteStall(StallInterface* wbm_stall) {
+bool WriteBufferManager::MaybeBeginWriteStall(StallInterface* wbm_stall) {
   assert(wbm_stall != nullptr);
-  assert(allow_stall_);
-
-  // Allocate outside of the lock.
   std::list<StallInterface*> new_node = {wbm_stall};
-
-  {
-    std::unique_lock<std::mutex> lock(mu_);
-    // Verify if the stall conditions are stil active.
-    if (ShouldStall()) {
-      stall_active_.store(true, std::memory_order_relaxed);
-      queue_.splice(queue_.end(), std::move(new_node));
-    }
-  }
-
-  // If the node was not consumed, the stall has ended already and we can signal
-  // the caller.
-  if (!new_node.empty()) {
-    new_node.front()->Signal();
```

Comment:
Cleanup4: I don't see why we need to signal wbm_stall here since if `wbm_stall` has been blocked due to another writer thread of the same DB has set it to blocked, that writer thread would be responsible for unclocking/signaling this wbm_stall not the current one.